src/DataTable/column.ts

```diff
@@ -0,0 +1,54 @@
+import {ObjectPaths, ObjectPathValue} from './utils'
+import {UniqueRow} from './row'
+import {SortStrategies, SortStrategy} from './sorting'
+
+export interface Column<Data extends UniqueRow> {
+  id?: string | undefined
+
+  /**
+   * Provide the name of the column. This will be rendered as a table header
+   * within the table itself
+   */
+  header: string
+
+  /**
+   * Optionally provide a field to render for this column. This may be the key
+   * of the object or a string that accesses nested objects through `.`. For
+   * exmaple: `field: a.b.c`
+   *
+   * Alternatively, you may provide a `renderCell` for this column to render the
+   * field in a row
+   */
+  field: ObjectPaths<Data>
+
+  /**
+   * Provide a custom component or render prop to render the data for this
+   * column in a row
+   */
+  renderCell?: ((data: Data) => React.ReactNode) | undefined
+
+  /**
+   * Specify if the value of this column for a row should be treated as a row
+   * header
+   */
+  rowHeader?: boolean | undefined
+
+  /**
+   * Specify if the table should sort by this column and, if applicable, a
+   * specific sort strategy or custom sort strategy
+   */
+  sortBy?: boolean | SortStrategies | SortStrategy<ObjectPathValue<Data, ObjectPaths<Data>>>
+}
+
+export function createColumnHelper<T extends UniqueRow>() {
```

Comment:
@mperrotti I think it would be helpful. It seems like it's more common for columns to not be defined inlined when looking through existing tables in dotcom (they use a `createTableMetadata` helper if I remember right)


Let me know if there would be a better way to do this, I tried using `Array<Column<T>>` but there is a normalization step that would be missing for the `id` field which lead to this helper

Reply:
>  It seems like it's more common for columns to not be defined inlined

Great point! I can't think of anything better than this.